components/sync15/src/enc_payload.rs

```diff
@@ -0,0 +1,110 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use crate::error;
+use crate::key_bundle::KeyBundle;
+use lazy_static::lazy_static;
+use serde::{Deserialize, Serialize};
+
+/// A representation of an encrypted payload. Used as the payload in EncryptedBso and
+/// also anywhere else the sync keys might be used to encrypt/decrypt, such as send-tab payloads.
+#[derive(Deserialize, Serialize, Clone, Debug)]
+pub struct EncryptedPayload {
+    #[serde(rename = "IV")]
+    pub iv: String,
+    pub hmac: String,
+    pub ciphertext: String,
+}
+
+impl EncryptedPayload {
+    #[inline]
+    pub fn serialized_len(&self) -> usize {
+        (*EMPTY_ENCRYPTED_PAYLOAD_SIZE) + self.ciphertext.len() + self.hmac.len() + self.iv.len()
```

Comment:
Is this always accurate?  I think that assumes there's zero overhead when JSON-encoding the strings.  What if one of them contain a `"` char that needs to be escaped to `\"`?

I'm guessing this code isn't new, but I figured it was worth flagging.

Reply:
I don't think the encrypted payload can actually have those chars? But maybe worth digging in to this to be sure, but let's not derail this patch with that because as you note, this code has existed for years

Reply:
I did a quick check and it looks like it's base64 encoded, so I don't think there's any chance of an issue.